extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableLocation.java

```diff
@@ -0,0 +1,125 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.location;
+
+import io.deephaven.api.SortColumn;
+import io.deephaven.base.verify.Require;
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.table.BasicDataIndex;
+import io.deephaven.engine.table.impl.locations.*;
+import io.deephaven.engine.table.impl.locations.impl.AbstractTableLocation;
+import io.deephaven.parquet.table.ParquetInstructions;
+import io.deephaven.parquet.table.location.ParquetTableLocation;
+import io.deephaven.parquet.table.location.ParquetTableLocationKey;
+import org.apache.iceberg.FileFormat;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class IcebergTableLocation implements TableLocation {
+
+    private static final String IMPLEMENTATION_NAME = IcebergTableLocation.class.getSimpleName();
+
+    private final ImmutableTableKey tableKey;
+    private final ImmutableTableLocationKey tableLocationKey;
+
+    AbstractTableLocation internalTableLocation;
+
+    public IcebergTableLocation(@NotNull final TableKey tableKey,
+            @NotNull final IcebergTableLocationKey tableLocationKey,
+            @NotNull final Object readInstructions) {
+        this.tableKey = Require.neqNull(tableKey, "tableKey").makeImmutable();
+        this.tableLocationKey = Require.neqNull(tableLocationKey, "tableLocationKey").makeImmutable();
+
+        if (tableLocationKey.format == FileFormat.PARQUET) {
+            this.internalTableLocation = new ParquetTableLocation(tableKey,
+                    (ParquetTableLocationKey) tableLocationKey.internalTableLocationKey,
+                    (ParquetInstructions) readInstructions);
+        } else {
+            throw new IllegalArgumentException("Unsupported file format: " + tableLocationKey.format);
+        }
+    }
+
+    @Override
+    public String getImplementationName() {
+        return IMPLEMENTATION_NAME;
+    }
+
+    @Override
+    @NotNull
+    public ImmutableTableKey getTableKey() {
+        return tableKey;
+    }
+
+    @Override
+    public @NotNull ImmutableTableLocationKey getKey() {
+        return tableLocationKey;
+    }
```

Comment:
You can make it consistent with `getTableKey()`
```suggestion
    @Override
    @NotNull
    public ImmutableTableLocationKey getKey() {
        return tableLocationKey;
    }
```